src/main/scala/se/nimsa/dcm4che/streams/DicomParsing.scala

```diff
@@ -55,6 +72,82 @@ trait DicomParsing {
     }
   }
 
+  // parse dicom UID data element from buffer
+  def parseUIDDataElement(data: ByteString, explicitVR: Boolean, assumeBigEndian: Boolean): DataElement = {
+    val maybeHeader = if (explicitVR) {
+      readHeaderExplicitVR(data, assumeBigEndian)
+    } else {
+      readHeaderImplicitVR(data)
+    }
+
+    if (maybeHeader.isEmpty) {
+      throw new DicomStreamException("Could not parse DICOM data element from stream.")
+    }
+
+    val (tag, vr, headerLength, length) = maybeHeader.get
+    val value = data.drop(headerLength).take(length)
+    DataElement(tag, vr, length, valueWithoutPadding(value))
+  }
+
+
+  def valueWithoutPadding(value: ByteString) =
+    if (value.takeRight(1).contains(0.toByte)) {
+      value.dropRight(1)
+    } else {
+      value
+    }
+
+  /**
+    * Read header of data element for explicit VR
+    * @param buffer current buffer
+    * @param assumeBigEndian true if big endian, false otherwise
+    * @return
+    */
+  def readHeaderExplicitVR(buffer: ByteString, assumeBigEndian: Boolean): Option[(Int, VR, Int, Int)] = {
```

Comment:
Now that these methods have been refactored from DicomValidateFlow to DicomParsing, can they be used by DicomPartFlow as well to cut code?